client/src/util/locationConverter.ts

```diff
@@ -0,0 +1,60 @@
+import { COORDINATE_PRECISION } from '../components/SearchBar/constants.ts';
+import { Location } from '../gql/graphql.ts';
+
+const DOUBLE_PATTERN = '-{0,1}\\d+(\\.\\d+){0,1}';
+
+const LAT_LON_PATTERN = '(' + DOUBLE_PATTERN + ')(\\s*,\\s*|\\s+)(' + DOUBLE_PATTERN + ')';
+
+const ID_SEPARATOR = ':';
+
+const NAME_SEPARATOR = '::';
+
+export function parseLocation(value: string): Location | null {
```

Comment:
I don't understand what this function does. In particular, there seems to be something going on with "name" which seems out of scope.

Reply:
These function is the typescript version of the[ LocationStringParser.java](https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/application/src/main/java/org/opentripplanner/api/common/LocationStringParser.java) class.

I removed the "name" parsing part.